src/services/UserService.test.ts

```diff
@@ -0,0 +1,265 @@
+import sinon from "sinon";
+import { stubInterface } from "ts-sinon";
+import NotFoundException from "../exceptions/NotFoundException";
+import User, { UserRole }  from "../entities/User";
+import getUserMock from "../mocks/UserMock";
+import IRepository from "../types/IRepository";
+import UserService from "./UserService";
+
+
+describe(
+  "UserService", () => {
+    const sandbox = sinon.createSandbox();
+    afterEach(() => sandbox.restore());
+
+    /* GET REQUESTS */
+
+    /* Get By Id Success */
+
+    it(
+      "getById success", async () => {
+        // Given
+        const userMock = getUserMock();
+        const fakeRepo = stubInterface<IRepository<User>>();
+
+        // When
+
+        fakeRepo.findOneBy.resolves(userMock);
```

Comment:
This is the problem behind your test failures. You are mocking the wrong method. In `UserService` you are using `findById`, but you are mocking `findOneBy` here. Therefore, when you are running this test and calling `new UserService().getById` it will actually try to run `repo.findById` which _is not mocked_ and therefore you end up having the unhandled promise exception.